core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -7487,6 +7487,21 @@ private void checkSemiJoinRuleOnAntiJoin(RelOptRule rule) {
         .checkUnchanged();
   }
 
+  @Test void testRecursiveQuery() {
+    final String sql = "WITH RECURSIVE aux(i) AS (\n"
+        + "  VALUES (1)\n"
+        + "  UNION ALL\n"
+        + "  SELECT i+1 FROM aux WHERE i < 10\n"
+        + ")\n"
+        + "SELECT * FROM aux";
+
+    sql(sql)
+        .withLateDecorrelate(true)
+        .withTrim(true)
+        .withRule() // empty program
+        .checkUnchanged();
+  }
+
```

Comment:
We are not testing any rules here so the test seems misplaced. Consider removing it.

Reply:
This seems like a test that should be added in `SqlToRelConverterTest` class.